src/rules/javascript/complex_prefs_defaults.js

```diff
@@ -0,0 +1,26 @@
+import { COMPLEX_PREFS_DEFAULTS } from 'messages/javascript';
+
+export default function(context) {
+  var filename = context.getFilename();
+
+  if (filename.indexOf('defaults/preferences/') === 0) {
```

Comment:
Yeah I think it sounds like we need a separate issue for making file name passing more robust especially if we're relying on it for narrowing tests down.